contrib/ruby/lib/trilogy.rb

```diff
@@ -28,34 +109,38 @@ def query_with_flags(sql, flags)
   ensure
     self.query_flags = old_flags
   end
-end
 
-Trilogy::Result.class_eval do
-  def count
-    rows.count
-  end
+  class Result
+    attr_reader :fields, :rows, :query_time
```

Comment:
Just noticed these warnings in the activerecord-trilogy-adapter tests:

```
warning: method redefined; discarding old fields
warning: method redefined; discarding old rows
warning: method redefined; discarding old query_time
```

Defining these in Ruby seems good to me, but then I assume we'll need to remove the `rb_define_attr`s.

Reply:
I'll look into this! 👀

Reply:
Done!